==13764== Memcheck, a memory error detector
==13764== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==13764== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==13764== Command: /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance
==13764== 
==13764== Conditional jump or move depends on uninitialised value(s)
==13764==    at 0x689E9D: wasmparser::validator::Validator::end (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x67F99C: wasmparser::validator::Validator::payload (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x67EC11: wasmparser::validator::Validator::validate_all (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2980DC: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x263778: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::validate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D44: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764==    by 0x25E051: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==  Uninitialised value was created by a stack allocation
==13764==    at 0x297FF0: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 
==13764== HEAP SUMMARY:
==13764==     in use at exit: 31,776 bytes in 103 blocks
==13764==   total heap usage: 542 allocs, 439 frees, 104,468 bytes allocated
==13764== 
==13764== 4 bytes in 1 blocks are still reachable in loss record 1 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x660210: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x657806: wasmer_compiler::translator::sections::parse_function_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x664CA4: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764== 
==13764== 7 bytes in 1 blocks are still reachable in loss record 2 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x65BD8F: wasmer_compiler::translator::environ::ModuleEnvironment::declare_func_export (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x658A5B: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x66522D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764== 
==13764== 8 bytes in 1 blocks are still reachable in loss record 3 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x661001: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x656CBF: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x664FBF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764== 
==13764== 8 bytes in 1 blocks are still reachable in loss record 4 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x661001: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x656D14: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x664FBF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764== 
==13764== 8 bytes in 1 blocks are still reachable in loss record 5 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2E282A: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x29E571: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x282B98: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 8 bytes in 1 blocks are indirectly lost in loss record 6 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x261B16: <alloc::boxed::Box<[T]> as core::clone::Clone>::clone (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x27EF7A: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57A0FD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57F8F2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D7A9: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764==    by 0x25E051: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 8 bytes in 1 blocks are indirectly lost in loss record 7 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x261A26: <alloc::boxed::Box<[T]> as core::clone::Clone>::clone (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x27EFA7: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57A0FD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57F8F2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D7A9: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764==    by 0x25E051: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 9 bytes in 1 blocks are still reachable in loss record 8 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x65C20F: wasmer_compiler::translator::environ::ModuleEnvironment::declare_function_name (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65A202: wasmer_compiler::translator::sections::parse_name_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x665420: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764== 
==13764== 32 bytes in 1 blocks are still reachable in loss record 9 of 43
==13764==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==13764==    by 0x4E3F165: dlsym (dlsym.c:70)
==13764==    by 0x73A5A3: fetch (weak.rs:61)
==13764==    by 0x73A5A3: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==13764==    by 0x73A5A3: min_stack_size (thread.rs:449)
==13764==    by 0x73A5A3: std::sys::unix::thread::Thread::new (thread.rs:49)
==13764==    by 0x2DB8C6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DEC46: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF7FF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 48 bytes in 1 blocks are still reachable in loss record 10 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x660117: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x656AC7: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x664FBF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764== 
==13764== 52 bytes in 1 blocks are possibly lost in loss record 11 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x653FE2: hashbrown::raw::inner::RawTable<T>::fallible_with_capacity (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x6539FE: _ZN9hashbrown3raw5inner17RawTable$LT$T$GT$14reserve_rehash17he29be5fc862d84d8E.llvm.8588116468508767673 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x658B91: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x66522D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 64 bytes in 4 blocks are still reachable in loss record 12 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2E4E2C: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E5CA7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF0D6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 64 bytes in 4 blocks are still reachable in loss record 13 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x73A4FC: alloc (alloc.rs:80)
==13764==    by 0x73A4FC: alloc (alloc.rs:174)
==13764==    by 0x73A4FC: exchange_malloc (alloc.rs:277)
==13764==    by 0x73A4FC: std::sys::unix::thread::Thread::new (thread.rs:44)
==13764==    by 0x2DB8C6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DEC46: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF7FF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 72 bytes in 1 blocks are still reachable in loss record 14 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x6416DD: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x64534B: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x284C55: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x27EF35: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57A0FD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57F8F2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D7A9: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764==    by 0x25E051: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 88 bytes in 1 blocks are still reachable in loss record 15 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x6635BF: hashbrown::raw::RawTable<T>::reserve_rehash (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x663962: hashbrown::raw::RawTable<T>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x652B74: hashbrown::map::HashMap<K,V,S>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65C294: wasmer_compiler::translator::environ::ModuleEnvironment::declare_function_name (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65A202: wasmer_compiler::translator::sections::parse_name_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x665420: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 120 bytes in 1 blocks are still reachable in loss record 16 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x660017: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x6588E3: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x66522D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x65B1D8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2826B4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764== 
==13764== 128 bytes in 4 blocks are still reachable in loss record 17 of 43
==13764==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13764==    by 0x706B2E: register_dtor (fast_thread_local.rs:33)
==13764==    by 0x706B2E: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==13764==    by 0x706B2E: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==13764==    by 0x729329: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==13764==    by 0x729329: __getit (local.rs:175)
==13764==    by 0x729329: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==13764==    by 0x729329: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==13764==    by 0x729329: std::sys_common::thread_info::set (thread_info.rs:40)
==13764==    by 0x2DBA4C: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x73A919: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13764==    by 0x73A919: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13764==    by 0x73A919: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13764==    by 0x52516DA: start_thread (pthread_create.c:463)
==13764==    by 0x57A271E: clone (clone.S:95)
==13764== 
==13764== 128 bytes in 4 blocks are still reachable in loss record 18 of 43
==13764==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13764==    by 0x2ED4FD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E4FD2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DC8D6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E0519: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DE62A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E10C0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DBA8E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x73A919: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13764==    by 0x73A919: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13764==    by 0x73A919: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13764==    by 0x52516DA: start_thread (pthread_create.c:463)
==13764==    by 0x57A271E: clone (clone.S:95)
==13764== 
==13764== 160 bytes in 4 blocks are still reachable in loss record 19 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2E2ABC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E011B: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E5F08: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1050: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF17A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 160 bytes in 4 blocks are still reachable in loss record 20 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2E2ABC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E0146: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E5F08: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1050: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF17A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 160 bytes in 4 blocks are still reachable in loss record 21 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2E2ABC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E60CB: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF1F0: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 160 bytes in 4 blocks are still reachable in loss record 22 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x707ECE: alloc (alloc.rs:80)
==13764==    by 0x707ECE: alloc (alloc.rs:174)
==13764==    by 0x707ECE: exchange_malloc (alloc.rs:277)
==13764==    by 0x707ECE: new<()> (mutex.rs:223)
==13764==    by 0x707ECE: std::thread::Thread::new (mod.rs:1149)
==13764==    by 0x2DB7B1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DEC46: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF7FF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 160 bytes in 4 blocks are still reachable in loss record 23 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2DB7D3: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DEC46: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF7FF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 192 bytes in 4 blocks are still reachable in loss record 24 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x724FC6: alloc (alloc.rs:80)
==13764==    by 0x724FC6: alloc (alloc.rs:174)
==13764==    by 0x724FC6: exchange_malloc (alloc.rs:277)
==13764==    by 0x724FC6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13764==    by 0x2E0126: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E5F08: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1050: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF17A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 192 bytes in 4 blocks are still reachable in loss record 25 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x724FC6: alloc (alloc.rs:80)
==13764==    by 0x724FC6: alloc (alloc.rs:174)
==13764==    by 0x724FC6: exchange_malloc (alloc.rs:277)
==13764==    by 0x724FC6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13764==    by 0x2E0151: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E5F08: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1050: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF17A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 192 bytes in 4 blocks are still reachable in loss record 26 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x724FC6: alloc (alloc.rs:80)
==13764==    by 0x724FC6: alloc (alloc.rs:174)
==13764==    by 0x724FC6: exchange_malloc (alloc.rs:277)
==13764==    by 0x724FC6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13764==    by 0x2E60D1: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF1F0: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 192 bytes in 4 blocks are still reachable in loss record 27 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x707F32: alloc (alloc.rs:80)
==13764==    by 0x707F32: alloc (alloc.rs:174)
==13764==    by 0x707F32: exchange_malloc (alloc.rs:277)
==13764==    by 0x707F32: new (condvar.rs:130)
==13764==    by 0x707F32: std::thread::Thread::new (mod.rs:1150)
==13764==    by 0x2DB7B1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DEC46: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF7FF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 320 bytes in 4 blocks are still reachable in loss record 28 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x707F81: alloc (alloc.rs:80)
==13764==    by 0x707F81: alloc (alloc.rs:174)
==13764==    by 0x707F81: exchange_malloc (alloc.rs:277)
==13764==    by 0x707F81: new<std::thread::Inner> (sync.rs:324)
==13764==    by 0x707F81: std::thread::Thread::new (mod.rs:1145)
==13764==    by 0x2DB7B1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DEC46: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF7FF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 384 bytes in 1 blocks are still reachable in loss record 29 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2E42DC: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E100C: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF17A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 416 bytes in 4 blocks are still reachable in loss record 30 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2DB868: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DEC46: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF7FF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 432 (416 direct, 16 indirect) bytes in 1 blocks are definitely lost in loss record 31 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x6A916D: wasmer_vm::instance::InstanceHandle::allocate_instance (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x27E901: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57A0FD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57F8F2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D7A9: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764==    by 0x25E051: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 512 bytes in 1 blocks are still reachable in loss record 32 of 43
==13764==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x72AA39: aligned_malloc (alloc.rs:95)
==13764==    by 0x72AA39: alloc (alloc.rs:22)
==13764==    by 0x72AA39: __rdl_alloc (alloc.rs:304)
==13764==    by 0x2E45F6: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF1AA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 640 bytes in 1 blocks are still reachable in loss record 33 of 43
==13764==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x72AA39: aligned_malloc (alloc.rs:95)
==13764==    by 0x72AA39: alloc (alloc.rs:22)
==13764==    by 0x72AA39: __rdl_alloc (alloc.rs:304)
==13764==    by 0x2DF3E6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 640 bytes in 1 blocks are still reachable in loss record 34 of 43
==13764==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x72AA39: aligned_malloc (alloc.rs:95)
==13764==    by 0x72AA39: alloc (alloc.rs:22)
==13764==    by 0x72AA39: __rdl_alloc (alloc.rs:304)
==13764==    by 0x2ED902: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2ED59D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E4FD2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DC8D6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E0519: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DE62A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E10C0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DBA8E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 664 bytes in 1 blocks are still reachable in loss record 35 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x282A23: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x26392A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x579D9F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BDE1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D659: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764==    by 0x25E051: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 808 bytes in 1 blocks are still reachable in loss record 36 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x64CC3F: alloc::collections::btree::map::BTreeMap<K,V>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x645751: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x284C55: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x27EF35: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57A0FD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x57F8F2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25D7A9: instance::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x25BE82: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x72BFD2: {{closure}} (rt.rs:52)
==13764==    by 0x72BFD2: do_call<closure-0,i32> (panicking.rs:348)
==13764==    by 0x72BFD2: try<i32,closure-0> (panicking.rs:325)
==13764==    by 0x72BFD2: catch_unwind<closure-0,i32> (panic.rs:394)
==13764==    by 0x72BFD2: std::rt::lang_start_internal (rt.rs:51)
==13764==    by 0x25E051: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 1,152 bytes in 4 blocks are possibly lost in loss record 37 of 43
==13764==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==13764==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==13764==    by 0x5252227: allocate_stack (allocatestack.c:627)
==13764==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==13764==    by 0x73A64C: std::sys::unix::thread::Thread::new (thread.rs:66)
==13764==    by 0x2DB8C6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DEC46: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF7FF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 1,520 bytes in 1 blocks are still reachable in loss record 38 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2DF242: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x29E571: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x282B98: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 1,536 bytes in 4 blocks are still reachable in loss record 39 of 43
==13764==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x72AA39: aligned_malloc (alloc.rs:95)
==13764==    by 0x72AA39: alloc (alloc.rs:22)
==13764==    by 0x72AA39: __rdl_alloc (alloc.rs:304)
==13764==    by 0x2E4ED0: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E5CA7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF0D6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 2,008 bytes in 1 blocks are still reachable in loss record 40 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2ED04E: crossbeam_epoch::sync::queue::Queue<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2ED8D2: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2ED59D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E4FD2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DC8D6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E0519: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DE62A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E10C0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DBA8E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x73A919: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13764==    by 0x73A919: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13764==    by 0x73A919: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13764== 
==13764== 4,096 bytes in 4 blocks are still reachable in loss record 41 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2E4E10: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E5CA7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DF0D6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E27F6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x725679: std::sync::once::Once::call_inner (once.rs:416)
==13764==    by 0x2DEDC0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E1BBB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x28ADA1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x291945: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2960F8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2B1B41: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764== 
==13764== 6,080 bytes in 4 blocks are still reachable in loss record 42 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2DDFF8: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E10C0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DBA8E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x73A919: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13764==    by 0x73A919: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13764==    by 0x73A919: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13764==    by 0x52516DA: start_thread (pthread_create.c:463)
==13764==    by 0x57A271E: clone (clone.S:95)
==13764== 
==13764== 8,160 bytes in 4 blocks are still reachable in loss record 43 of 43
==13764==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13764==    by 0x2EA92F: crossbeam_epoch::internal::Local::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2ED524: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E4FD2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DC8D6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E0519: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DE62A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2E10C0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x2DBA8E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/instance)
==13764==    by 0x73A919: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13764==    by 0x73A919: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13764==    by 0x73A919: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13764==    by 0x52516DA: start_thread (pthread_create.c:463)
==13764==    by 0x57A271E: clone (clone.S:95)
==13764== 
==13764== LEAK SUMMARY:
==13764==    definitely lost: 416 bytes in 1 blocks
==13764==    indirectly lost: 16 bytes in 2 blocks
==13764==      possibly lost: 1,204 bytes in 5 blocks
==13764==    still reachable: 30,140 bytes in 95 blocks
==13764==         suppressed: 0 bytes in 0 blocks
==13764== 
==13764== For counts of detected and suppressed errors, rerun with: -v
==13764== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: 0 from 0)
